### PR TITLE
Update manifest with new security references 

### PR DESCRIPTION
**Pull request by scrawfor99**
### Description
Update the manifest files with the new references for Security and Security dashboards repos. 

### Issues Resolved

(https://github.com/opensearch-project/security/issues/3028)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: manifests/2.10.0/opensearch-2.10.0.yml [hunk 1]
```diff
@@ -52,7 +52,7 @@ components:
       - gradle:dependencies:opensearch.version
   - name: security
     repository: https://github.com/opensearch-project/security.git
-    ref: '2.10'
+    ref: tags/2.10.0.0
```
Comment (gaiksaya) [inline manifests/2.10.0/opensearch-2.10.0.yml:55]: tags cannot be added before the release. Same for another change

**Review by gaiksaya — CHANGES_REQUESTED** [review]

File: manifests/2.10.0/opensearch-2.10.0.yml (hunk 1 above)
Comment (scrawfor99) [reply manifests/2.10.0/opensearch-2.10.0.yml:55]: We should remove this step from the "pre-release" section of the release instructions then: https://github.com/opensearch-project/security/issues/3028#issue-1810426686. 

The last item reads as updating the tags before the release. Perhaps this is not what is meant?

Comment (gaiksaya) [reply manifests/2.10.0/opensearch-2.10.0.yml:55]: Can you point me where it said tags?

Comment (scrawfor99) [reply manifests/2.10.0/opensearch-2.10.0.yml:55]: I was referring to `Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) for the next developer iteration.` 

I am not sure what that means that is separate from the release branch update--so I figured it meant the tags.

Comment (gaiksaya) [reply manifests/2.10.0/opensearch-2.10.0.yml:55]: Next developer iteration is 2.11.0. So once you bump the version in 2.x branch, add your component to next manifest. The PR is still in review: https://github.com/opensearch-project/opensearch-build/pull/3969

Comment (scrawfor99) [reply manifests/2.10.0/opensearch-2.10.0.yml:55]: Gotcha, my misunderstanding.

**Pull request closed without merging**